rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -262,7 +262,9 @@ PublisherBase::default_incompatible_qos_callback(
   std::string policy_name = qos_policy_name_from_kind(event.last_policy_kind);
   RCLCPP_WARN(
     rclcpp::get_logger(rcl_node_get_logger_name(rcl_node_handle_.get())),
-    "New subscription discovered on this topic, requesting incompatible QoS. "
+    "New subscription discovered on topic %s, requesting incompatible QoS. "
```

Comment:
```suggestion
    "New subscription discovered on topic '%s', requesting incompatible QoS. "
```

Same thing below.

Reply:
> I think I need to also go find where CI is comparing the output in a test case, and update the expected output that is baked into the test case.

Yeah, this will definitely need to be done.  The PR job passes because we skip this test on Fast-DDS, but this will fail for Cyclone and Connext.